tests/lib/retry.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+snap_install() {
```

Comment:
Maybe we could express in the name the fact that retries are involved, `snap_retry_install` ?